packages/lab/stories/dialog/dialog.doc.stories.mdx

```diff
@@ -0,0 +1,155 @@
+
+#### Initial focus
+
+You can control which element receives focus when the Dialog is opened by passing a `initialFocus` prop to the Dialog component. This prop accepts a number, which is the index of all tabbable elements within the Dialog that should receive focus. If no element is found, focus will be set to the first focusable element within the Dialog. The `initialFocus` can also be a ref to an element within the Dialog.
+
+## Alert dialogs
+
+For alert dialogs you can set the `status` prop to `error`, `warning`, `success`, or `info` to render a status icon in the title of the Dialog. For accessibility reasons, you should pass a `role="alertdialog"` to the Dialog component when rendering an alert dialog.
+
+### Error
+
+Use this Dialog to communicate a system error or an error relating to the user's current task, which fundamentally prevents them from continuing or completing it.
+An error icon displays alongside a title and supporting message.
+
+<Canvas>
+  <Story inline={false} height="500px" id="lab-dialog--error-status" />
+</Canvas>
+
+### Warning
+
+You can warn users of an issue that's related to their current task. Although the issue does not prevent the user from continuing or completing their task, it may have repercussions or an indirect impact elsewhere. This Dialog is displayed with a warning icon.
+
+<Canvas>
+  <Story inline={false} height="500px" id="lab-dialog--warning-status" />
+</Canvas>
+
+### Info
+
+<Canvas>
+  <Story inline={false} height="500px" id="lab-dialog--info-status" />
+</Canvas>
+
+### Success
+
+Use this Dialog to confirm that a user's action has been completed successfully. This Dialog is displayed with a success icon.
+
+<Canvas>
+  <Story inline={false} height="500px" id="lab-dialog--success-status" />
+</Canvas>
+
+## Content dialogs
+
+### Accent
+
+For content dialogs, you can pass the `accent` prop to `DialogTitle` which accepts a boolean to render a accent bar in the title of the Dialog. You cannot render the accent bar when a status is also being rendered.
+
+<Canvas>
+  <Story inline={false} height="500px" id="lab-dialog--accent" />
+</Canvas>
+
+### Dialog Close Button
+
+For content dialogs, you can render the `DialogCloseButton` as a child of the `Dialog` component to render a close button in the top right cornder of the Dialog. You can pass the `onClick` prop like with any `Button` component to handle the close button click.
+
+```tsx
+<Dialog open={open} onOpenChange={handleOpenChange}>
+  <DialogTitle>Dialog title</DialogTitle>
+  <DialogContent>Dialog content</DialogContent>
+  <DialogActions>
+    <Button variant="cta">OK</Button>
+  </DialogActions>
+  <DialogCloseButton onClick={handleCloseDialog} />
+</Dialog>
+```
+
+### Scroll
+
+If the content of the Dialog makes it so that the Dialog streaches to the end of the screen, the `DialogContent` area will become scrollable.
```

Comment:
```suggestion
If the content of the Dialog makes it so that the Dialog stretches to the end of the screen, the `DialogContent` area will become scrollable.
```